### PR TITLE
new api endpoint ip and mac

### PR DESCRIPTION
**Pull request by fadkeabhi**
added api endpoint foe random ip address and random mac address

### CODE REVIEW

File: modules/ip/api/ip-routes.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    var macAddress = "";
+    for (var i = 0; i < 6; i++) {
+        macAddress+=hexDigits.charAt(Math.round(Math.random() * 15));
+        macAddress+=hexDigits.charAt(Math.round(Math.random() * 15));
+        if (i != 5) macAddress += ":";
+    }
+    return macAddress;
+}
+
+    app.get("/ip/", (req: Request, res: Response) => {
```
Comment (ageddesi) [inline modules/ip/api/ip-routes.ts:15]: Thank you for this, we no longer directly edit swagger files instead we use OpenAPI comments in the api route file, please can you update this.

File: modules/ip/api/ip-routes.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { Request, Response } from 'express';
+import * as core from 'express-serve-static-core';
+
+function genMAC(){
```
Comment (ageddesi) [inline modules/ip/api/ip-routes.ts:4]: Can you move this function into a file under  a util folder in the module

File: modules/ip/api/ip-routes.ts (hunk 1 above)
Comment (ageddesi) [reply modules/ip/api/ip-routes.ts:15]: We also like to give the user the ability to define how many they would like in return, please can you add a qty param to the url and use the util function getQtyFromRequest to get the qty

**Review by ageddesi — CHANGES_REQUESTED** [review]
Thank you for your contribution, there is a few change requests and also we would appreciate it if you can add tests for the work.

File: modules/ip/api/ip-routes.ts (hunk 1 above)
Comment (fadkeabhi) [reply modules/ip/api/ip-routes.ts]: added OpenAPI comments in the api route file, also added qty parameter in ip and macusing function  getQtyFromRequest to get the qty

File: modules/ip/api/ip-routes.ts (hunk 2 above)
Comment (fadkeabhi) [reply modules/ip/api/ip-routes.ts]: moved the function to util folder of respective module

**Review by ageddesi — APPROVED** [review]

**Comment by ageddesi** [comment]
Thanks @fadkeabhi for the hard work on this contribution. Happy Hacktoberfest 👍 

@all-contributors add @fadkeabhi code

**Comment by ageddesi** [comment]
https://github.com/all-contributors add @fadkeabhi code

**Comment by ageddesi** [comment]
@all-contributors add @fadkeabhi code

**Pull request merged**